src/sources/groups.js

```diff
@@ -15,50 +14,55 @@ const MULTI_TENANT_BUCKET = "nextstrain-groups";
 
 
 class GroupSource extends Source {
-  constructor(groupOrName) {
+  constructor(name) {
     super();
+    this.name = name;
+  }
```

Comment:
I briefly mentioned this in the commit message of 82e3909318e6244a9c13dddc335721fdb198e721 but it's not obvious from the change itself. This is all because of circular dependencies. If both are to be set in their constructors, we can have either `Group.source` or `GroupSource.group`, but not both.

For example, errors will arise if you try to set `this.group` in the constructor here.

In CJS, this was a bit cryptic:

```
nextstrain.org/src/sources/groups.js:20
    this.group = groupOrName instanceof Group
                             ^

TypeError: Right-hand side of 'instanceof' is not an object
```

With ESM, it's more obvious:

```
RangeError: Maximum call stack size exceeded
    at normalizeGroupName (file://.../nextstrain.org/src/groups.js:214:15)
    at new Group (file://.../nextstrain.org/src/groups.js:43:43)
    at new GroupSource (file://.../nextstrain.org/src/sources/groups.js:23:9)
    at new Group (file://.../nextstrain.org/src/groups.js:58:19)
    at new GroupSource (file://.../nextstrain.org/src/sources/groups.js:23:9)
    at new Group (file://.../nextstrain.org/src/groups.js:58:19)
    at new GroupSource (file://.../nextstrain.org/src/sources/groups.js:23:9)
    at new Group (file://.../nextstrain.org/src/groups.js:58:19)
    at new GroupSource (file://.../nextstrain.org/src/sources/groups.js:23:9)
    at new Group (file://.../nextstrain.org/src/groups.js:58:19)
    at new GroupSource (file://.../nextstrain.org/src/sources/groups.js:23:9)
    at new Group (file://.../nextstrain.org/src/groups.js:58:19)
    at new GroupSource (file://.../nextstrain.org/src/sources/groups.js:23:9)
    at new Group (file://.../nextstrain.org/src/groups.js:58:19)
    at new GroupSource (file://.../nextstrain.org/src/sources/groups.js:23:9)
    at new Group (file://.../nextstrain.org/src/groups.js:58:19)
npm ERR! code ELIFECYCLE
```

Reply:
I refactored the ESM to have a top-level import which differs from the non-global `require` in `GroupSource.group` (didn't want to deal with non-global async `await import()`). This means circular dependency is an issue again, at least in the tests. Instead of trying to hack around it, I think there's an inherent issue with trying to cross-reference the two classes within each other. When I started on 1258b213335f4f1dd623f0b627b74c4de17f675b, my goal was the one-way referencing described in the title which would avoid circular dependency, but it's not trivial (and maybe not desirable) to remove `GroupSource.group`.

An alternative would be to revert to the original one-way referencing of GroupSource -> Group by dropping 1258b213335f4f1dd623f0b627b74c4de17f675b in favor of something like 9123008895a012a76323039266e93522291e72ee.

Reply:
> I think there's an inherent issue with trying to cross-reference the two classes within each other.

Hmm. There should be a way to do this. It's not an uncommon pattern for classes, though it's perhaps easier if the classes aren't in separate files.

> If both are to be set in their constructors, we can have either `Group.source` or `GroupSource.group`, but not both.

I don't think we need to set both in their constructors. `Group.source` can be a computed property. `GroupSource.group` should probably be a singleton, but could be lazily set.

Reply:
Oh!!! Thanks for making me think twice about this. Actually, these circular dependencies are [totally appropriate with ES6](http://calculist.org/blog/2012/06/29/static-module-resolution/#cyclic-dependencies). The actual app runs fine. The issue is with an older version of Jest not behaving well. Upgrading Jest fixes things: 6ef1ec2d45574f19d1bc4c9bd35c3f94e5bdbf31

Reply:
I modified and reworded/split 1258b213335f4f1dd623f0b627b74c4de17f675b into 833ea30...0af95cc. So this is a pretty minimal change now – adding the computed property `Group.source`.

Reply:
Awesome!